### PR TITLE
Add key events to filter bank

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
Adds the key events filter to the filter bank and provides a click handler to handle filtering by key events.

## Why?
Whilst in the automatic filter BETA, filtering by key events should move inside the filter bank rather than live as its own toggle. This is the case for both mobile and desktop. 

## Screenshots

| Desktop      | After      |
|-------------|------------|
| ![desktop][] | ![after][] |

[desktop]: https://user-images.githubusercontent.com/20416599/175982580-fe291173-43ed-4d90-a438-8264c711c657.png
[after]: https://example.com/after.png

### CODE REVIEW

File: dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		column-gap: 0;
+	}
+`;
+
+export const TopicFilterBank = ({
+	availableTopics,
+	selectedTopics,
+	format,
+	keyEvents,
+	keyEventsId,
```
Comment (joecowton1) [inline dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx:51]: As mentioned earlier, I think we'd only need this if we wanted to scroll to the button on click - i'd imagine we'd be better of setting `location.hash` to an id on the filter bank or even somewhere at the top of the liveblog maybe?

**Review by joecowton1 — COMMENTED** [review]
Looks good! most comments are just about not needing the `id`

File: dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx (hunk 1 above)
Comment (abeddow91) [reply dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx:51]: Got you, cheers Joe! I'll remove this

File: dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	filterKeyEvents = false,
+}: Props) => {
+	const handleClick = () => {
+		const urlParams = new URLSearchParams(window.location.search);
+		urlParams.delete('page'); // direct to the first page
+		urlParams.set('filterKeyEvents', `${!filterKeyEvents.toString()}`);
+		urlParams.delete('topics');
+		window.location.hash = keyEventsId;
```
[the remark this answers is not in the document]
Comment (abeddow91) [reply dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx:59]: As above

File: dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+export const TopicFilterBank = ({
+	availableTopics,
+	selectedTopics,
+	format,
+	keyEvents,
+	keyEventsId,
+	filterKeyEvents = false,
+}: Props) => {
+	const handleClick = () => {
```
[the remark this answers is not in the document]
Comment (abeddow91) [reply dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx:54]: Good point - I'll move to `handleKeyEventClick`

File: dotcom-rendering/src/web/components/TopicFilterBank.stories.tsx [hunk 4]
```diff
@@ -43,6 +50,15 @@ export const topicBank = () => {
...
 				format={format}
+				keyEvents={[
+					{
+						...baseProperties,
+						blockFirstPublished: 1638279933000,
+						title: 'title',
+					},
+				]}
+				filterKeyEvents={false}
+				keyEventsId={'key-events-filter-mobile'}
```
[the remark this answers is not in the document]
Comment (abeddow91) [reply dotcom-rendering/src/web/components/TopicFilterBank.stories.tsx:61]: 👍

File: dotcom-rendering/src/web/layouts/LiveLayout.tsx [hunk 5]
```diff
@@ -800,6 +800,15 @@ export const LiveLayout = ({ CAPIArticle, NAV, format }: Props) => {
 															CAPIArticle.selectedTopics
 														}
 														format={format}
+														keyEvents={
+															CAPIArticle.keyEvents
+														}
+														filterKeyEvents={
+															CAPIArticle.filterKeyEvents
+														}
+														keyEventsId={
```
[the remark this answers is not in the document]
Comment (abeddow91) [reply dotcom-rendering/src/web/layouts/LiveLayout.tsx:809]: 👍

File: dotcom-rendering/src/web/lib/LiveBlogRenderer.tsx [hunk 6]
```diff
@@ -110,6 +112,9 @@ export const LiveBlogRenderer = ({
 						availableTopics={availableTopics}
 						selectedTopics={selectedTopics}
 						format={format}
+						keyEvents={keyEvents}
+						filterKeyEvents={filterKeyEvents}
+						keyEventsId={'key-events-filter-mobile'}
```
[the remark this answers is not in the document]
Comment (abeddow91) [reply dotcom-rendering/src/web/lib/LiveBlogRenderer.tsx:117]: 👍

**Review by joecowton1 — APPROVED** [review]

**Pull request merged**